micrometer-core/src/main/java/io/micrometer/core/instrument/HighCardinalityTagsDetector.java

```diff
@@ -183,4 +192,120 @@ private static long calculateThreshold() {
         return Math.max(1_000, Math.min(allowance * 2_000, 2_000_000));
     }
 
+    /**
+     * Builder for {@code HighCardinalityTagsDetector}.
+     */
+    public static class Builder {
+
+        private final MeterRegistry registry;
+
+        private long threshold = calculateThreshold();
+
+        private Duration delay = DEFAULT_DELAY;
+
+        @Nullable private Consumer<String> highCardinalityMeterNameConsumer;
+
+        @Nullable private Consumer<HighCardinalityMeterInfo> highCardinalityMeterInfoConsumer;
+
+        /**
+         * Create a {@code Builder}.
+         * @param registry registry
+         */
+        public Builder(MeterRegistry registry) {
+            this.registry = registry;
+        }
+
+        /**
+         * Set threshold.
+         * @param threshold threshold
+         * @return this builder
+         */
+        public Builder threshold(long threshold) {
+            this.threshold = threshold;
+            return this;
+        }
+
+        /**
+         * Set delay.
+         * @param delay delay
+         * @return this builder
+         */
+        public Builder delay(Duration delay) {
+            this.delay = delay;
+            return this;
+        }
+
+        /**
+         * Set high cardinality meter name {@link Consumer}.
+         * @param highCardinalityMeterNameConsumer high cardinality meter name
+         * {@code Consumer}
+         * @return this builder
+         */
+        public Builder highCardinalityMeterNameConsumer(Consumer<String> highCardinalityMeterNameConsumer) {
+            this.highCardinalityMeterNameConsumer = highCardinalityMeterNameConsumer;
+            return this;
+        }
+
+        /**
+         * Set {@code HighCardinalityMeterInfo} {@link Consumer}.
+         * @param highCardinalityMeterInfoConsumer {@code HighCardinalityMeterInfo}
+         * {@code Consumer}
+         * @return this builder
+         */
+        public Builder highCardinalityMeterInfoConsumer(
+                Consumer<HighCardinalityMeterInfo> highCardinalityMeterInfoConsumer) {
+            this.highCardinalityMeterInfoConsumer = highCardinalityMeterInfoConsumer;
+            return this;
+        }
+
+        /**
+         * Build {@code HighCardinalityTagsDetector}.
+         * @return {@code HighCardinalityTagsDetector}
+         */
+        public HighCardinalityTagsDetector build() {
+            HighCardinalityTagsDetector highCardinalityTagsDetector = new HighCardinalityTagsDetector(this.registry,
+                    this.threshold, this.delay, this.highCardinalityMeterNameConsumer);
+            highCardinalityTagsDetector.meterInfoConsumer = this.highCardinalityMeterInfoConsumer;
```

Comment:
It's probably worth adding a constructor that takes the `Consumer<HighCardinalityMeterInfo>` and deprecating the one that takes `Consumer<String>`. Then the consumer field can be marked final. Was there any reason in particular to set it this way instead of a constructor?

Reply:
@shakuzen It would break existing lambda expressions from users and require casting to use lambda expressions for both.